common/channel/src/message_queues.rs

```diff
@@ -65,13 +66,9 @@ impl<K: Eq + Hash + Clone, T> PerKeyQueue<K, T> {
         max_queue_size_per_key: usize,
         counters: Option<&'static IntCounterVec>,
     ) -> Self {
-        assert!(
-            max_queue_size_per_key > 0,
-            "max_queue_size_per_key should be > 0"
-        );
         Self {
             queue_style,
-            max_queue_size: max_queue_size_per_key,
+            max_queue_size: NonZeroUsize::new(max_queue_size_per_key).unwrap(),
```

Comment:
@nikakis : Unfortunately, having an `unwrap()` here isn't much better than the assert. Will be better if we can change the `fn new(...)` to also take `NonZeroUsize` as an argument instead of `usize`.

Reply:
Yes, i understand. I will do the changes asap.
Thanks you for showing me the right direction @bothra90 !!!

Reply:
As far as I understand you want to assert the `max_queue_size_per_key` value during compile-time by using constants. Probably something like that : 
```
const fn new_non_zero_usize(value: usize)-> NonZeroUsize {
    unsafe{
        NonZeroUsize::new_unchecked(value)
    }
}

const MAX_QUEUE_SIZE_PER_KEY: NonZeroUsize = new_non_zero_usize(0);
```

Reply:
Actually, that wasn't what I meant. It's better to use `.unwrap()` in clients of libra_channel where the channel is constructed than having it hidden in the library. We don't need to add any unsafe code here.

Reply:
Ok, i think now i have a better picture. Sorry for asking all this but i am a newbiew here.
I am doing the changes now. 
Thanks again !!